spiped/dispatch.c

```diff
@@ -130,20 +163,37 @@ callback_gotconn(void * cookie, int s)
 	if ((sas = sock_addr_duplist(A->sas)) == NULL)
 		goto err1;
 
+	/* Create new conn_list_node. */
+	if ((node_new = malloc(sizeof(struct conn_list_node))) == NULL)
+		goto err2;
+	node_new->prev = NULL;
+	node_new->next = NULL;
+
 	/* Create a new connection. */
-	if (proto_conn_create(s, sas, A->decr, A->nofps, A->requirefps,
-	    A->nokeepalive, A->K, A->timeo, callback_conndied, A)) {
+	if ((node_new->conn_cookie = proto_conn_create(s, sas, A->decr,
+	    A->nofps, A->requirefps, A->nokeepalive, A->K, A->timeo,
+	    callback_conndied, A)) == NULL) {
 		warnp("Failure setting up new connection");
-		goto err2;
+		goto err3;
 	}
 
+	/* Link node_new to the beginning of the conn_cookies list. */
+	if (A->conn_cookies != NULL)
+		A->conn_cookies = node_new;
```

Comment:
This looks incorrect to me. If `A->conn_cookies` previously was `NULL` it would link `node_new` to itself.